xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/AbstractDocumentListClassPropertyValuesProvider.java

```diff
@@ -89,11 +92,19 @@ public PropertyValue getValue(ClassPropertyReference propertyReference, Object r
         }
 
         DocumentReference documentReference = this.documentReferenceResolver.resolve(reference, propertyReference);
-        PropertyValue propertyValue = super.getValue(propertyReference, documentReference);
-        if (propertyValue != null) {
-            propertyValue.setValue(reference);
+
+        PropertyValue propertyValue = null;
+        if (xcontext.getWiki().exists(documentReference, xcontext)) {
+            propertyValue = super.getValue(propertyReference, documentReference);
         }
 
+        if (propertyValue == null) {
+            propertyValue = new PropertyValue();
+        }
+
+        propertyValue.setValue(reference);
+        propertyValue.getMetaData().put("freeText", propertyDefinition.getFreeText());
```

Comment:
Why do you need this meta data? Where is it used? "freeText" is about the property itself (whether it supports free text or not) not about one value of a property (i.e. the value you're returning here).

Reply:
Thanks for checking, this metadata is useless, I forgot to remove this change.